### ISSUE TITLE
Feature request for GPU-assisted validation: print the detailed information for OOB buffer

### ISSUE THREAD

**Issue by jaebaek** [KhronosGroup/Vulkan-ValidationLayers#2267]
This is a feature request for GPU-assisted validation.

Since spirv-opt supports [OpenCL.DebugInfo.100](https://www.khronos.org/registry/spir-v/specs/unified1/OpenCL.DebugInfo.100.html), we can print what buffer causes the OOB error using the debug information (if it exists).

In addition, I found that the `OpLine` information for the instruction (the location in the high-level language source code) that causes the OOB error is not provided when I ran the GPU-assisted validation. I guess that the validation wants to support it but the implementation is not done .. am I right?
If it is not in the TODO list, I want to suggest it too.

**Comment by Tony-LunarG** [comment]
With respect to the OpLine information, the GPU-AV code does parse the shader source and print OpLine info when it's found.  For instance, running the validation layer test GpuValidationArrayOOBGraphicsShaders, the third subtest generates the following error message:
```
Validation Error: [ UNASSIGNED-Descriptor index out of bounds ] Object 0: handle = 0x19cb58d1a18, type = VK_OBJECT_TYPE_QUEUE; | MessageID = 0x68698324 | 
Index of 25 used to index descriptor array of length 6.  Command buffer (0x19cbf4b69e8). Draw Index 0. Pipeline (0x8f32de000000003b). 
Shader Module (0x9fe8bd0000000038). Shader Instruction Index = 102.  Stage = Vertex. 
Vertex Index = 1 Instance Index = 0.  Shader validation error occurred at line 11.
11:    gl_Position += 1e-30 * texture(tex[uniform_index_buffer.tex_index[0]], vec2(0, 0));
```
@jaebaek when you ran GPU-AV and didn't see OpLine information, are you certain that your shader was compiled debuggable?

**Comment by jaebaek** [comment]
I modified some unit tests and ran them but I did not know `GpuValidationArrayOOBGraphicsShaders` demonstrates it. Since all the unit tests pass without a failure, it might be fine.

One question .. what if the code has OpLine without the shader source?

**Comment by Tony-LunarG** [comment]
I believe you'll get an error that looks like "Unable to find SPIR-V OpString for file id " << reported_file_id << " from OpLine instruction."

**Comment by jaebaek** [comment]
Ah .. I remember I saw it. Yes, I think we can provide the file name and line/column numbers even when we do not have the source code. It will be very useful (at least, for me). What do you think?

**Comment by Tony-LunarG** [comment]
Yes, that seems reasonable.

**Comment by Tony-LunarG** [comment]
The OpLine portion of this request has been addressed by #2271

**Comment by mark-lunarg** [comment]
Added to the GPU-Assisted validation project page, [here](https://github.com/KhronosGroup/Vulkan-ValidationLayers/projects/3).
